From ab9eba5ac30d4dd5775554b6b9d263b2bd0aa7bb Mon Sep 17 00:00:00 2001
From: Aran Dalton <arda@allwinnertech.com>
Date: Thu, 26 Mar 2026 19:53:34 +0800
Subject: [PATCH 45/63] ANDROID: GKI: update symbol list for allwinner

5 function symbol(s) added
  'int hdmi_drm_infoframe_check(struct hdmi_drm_infoframe*)'
  'int ir_raw_gen_manchester(struct ir_raw_event**, unsigned int, const struct ir_raw_timings_manchester*, unsigned int, u64)'
  'int ir_raw_gen_pd(struct ir_raw_event**, unsigned int, const struct ir_raw_timings_pd*, unsigned int, u64)'
  'int ir_raw_handler_register(struct ir_raw_handler*)'
  'void ir_raw_handler_unregister(struct ir_raw_handler*)'

Bug: 496758254
Bug: 498402639
Change-Id: Ifaddbc0ce61c635b74ef1850dafb5f0f93edf9e6
Signed-off-by: Aran Dalton <arda@allwinnertech.com>
(cherry picked from commit 2141016c6437483f0cdc7b1e280961094efda35b)
---
 gki/aarch64/abi.stg           | 290 ++++++++++++++++++++++++++++++++++
 gki/aarch64/symbols/allwinner |   5 +
 2 files changed, 295 insertions(+)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index 90d68d4d6d07..73e94e6d0298 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -498,6 +498,11 @@ pointer_reference {
   kind: POINTER
   pointee_type_id: 0x2eabff79
 }
+pointer_reference {
+  id: 0x0144254e
+  kind: POINTER
+  pointee_type_id: 0x2f5073a5
+}
 pointer_reference {
   id: 0x014b8475
   kind: POINTER
@@ -17808,6 +17813,11 @@ pointer_reference {
   kind: POINTER
   pointee_type_id: 0x8a68c517
 }
+pointer_reference {
+  id: 0x281136d8
+  kind: POINTER
+  pointee_type_id: 0x8a043dfc
+}
 pointer_reference {
   id: 0x281af600
   kind: POINTER
@@ -21758,6 +21768,11 @@ pointer_reference {
   kind: POINTER
   pointee_type_id: 0x9e5ef2fc
 }
+pointer_reference {
+  id: 0x2d07cf05
+  kind: POINTER
+  pointee_type_id: 0x9e5fda8b
+}
 pointer_reference {
   id: 0x2d081f17
   kind: POINTER
@@ -28663,6 +28678,11 @@ pointer_reference {
   kind: POINTER
   pointee_type_id: 0xea5f49fa
 }
+pointer_reference {
+  id: 0x300812ba
+  kind: POINTER
+  pointee_type_id: 0xea60ac77
+}
 pointer_reference {
   id: 0x30092723
   kind: POINTER
@@ -34383,6 +34403,11 @@ pointer_reference {
   kind: POINTER
   pointee_type_id: 0xcd1e6a9f
 }
+pointer_reference {
+  id: 0x39daf541
+  kind: POINTER
+  pointee_type_id: 0xcd2b3398
+}
 pointer_reference {
   id: 0x39db4e33
   kind: POINTER
@@ -35413,6 +35438,11 @@ pointer_reference {
   kind: POINTER
   pointee_type_id: 0xc5ff6f1a
 }
+pointer_reference {
+  id: 0x3bf44f99
+  kind: POINTER
+  pointee_type_id: 0xc591d8f8
+}
 pointer_reference {
   id: 0x3bf581e3
   kind: POINTER
@@ -41413,6 +41443,11 @@ qualified {
   qualifier: CONST
   qualified_type_id: 0x4474c556
 }
+qualified {
+  id: 0xcd2b3398
+  qualifier: CONST
+  qualified_type_id: 0x44eb78eb
+}
 qualified {
   id: 0xcd2dde52
   qualifier: CONST
@@ -43318,6 +43353,11 @@ qualified {
   qualifier: CONST
   qualified_type_id: 0xd9402e22
 }
+qualified {
+  id: 0xea60ac77
+  qualifier: CONST
+  qualified_type_id: 0xd9c50754
+}
 qualified {
   id: 0xea647a11
   qualifier: CONST
@@ -72152,6 +72192,12 @@ member {
   type_id: 0x4585663f
   offset: 224
 }
+member {
+  id: 0xd8ab47be
+  name: "bit_pulse"
+  type_id: 0x4585663f
+  offset: 64
+}
 member {
   id: 0x57244bc9
   name: "bit_rate"
@@ -72169,6 +72215,12 @@ member {
   name: "bit_size"
   type_id: 0xe62ebf07
 }
+member {
+  id: 0xb6979874
+  name: "bit_space"
+  type_id: 0x7dc8196c
+  offset: 96
+}
 member {
   id: 0xa620dbc0
   name: "bitfield32_valid"
@@ -78685,6 +78737,12 @@ member {
   name: "card_no"
   type_id: 0xe62ebf07
 }
+member {
+  id: 0xb623a0ae
+  name: "carrier"
+  type_id: 0xc9082b19
+  offset: 320
+}
 member {
   id: 0xb623a769
   name: "carrier"
@@ -82254,6 +82312,12 @@ member {
   type_id: 0x92233392
   offset: 27200
 }
+member {
+  id: 0x458b8235
+  name: "clock"
+  type_id: 0x4585663f
+  offset: 64
+}
 member {
   id: 0x458b86d4
   name: "clock"
@@ -94798,6 +94862,12 @@ member {
   type_id: 0xe62ebf07
   offset: 4320
 }
+member {
+  id: 0x6e88de7a
+  name: "decode"
+  type_id: 0x2d07cf05
+  offset: 192
+}
 member {
   id: 0x86c5c02d
   name: "decode_vbi_line"
@@ -107391,6 +107461,12 @@ member {
   offset: 22
   bitsize: 1
 }
+member {
+  id: 0xf632952b
+  name: "encode"
+  type_id: 0x281136d8
+  offset: 256
+}
 member {
   id: 0x6c984f90
   name: "encode_fh"
@@ -129259,12 +129335,23 @@ member {
   type_id: 0x3b40190b
   offset: 320
 }
+member {
+  id: 0xe8a40f03
+  name: "header_pulse"
+  type_id: 0x4585663f
+}
 member {
   id: 0xf67296ed
   name: "header_size"
   type_id: 0x914dbfdc
   offset: 2816
 }
+member {
+  id: 0xd9e8634f
+  name: "header_space"
+  type_id: 0x4585663f
+  offset: 32
+}
 member {
   id: 0x61467847
   name: "headers"
@@ -140230,6 +140317,13 @@ member {
   type_id: 0x0baa70a7
   offset: 896
 }
+member {
+  id: 0xeac27b10
+  name: "invert"
+  type_id: 0x4585663f
+  offset: 96
+  bitsize: 1
+}
 member {
   id: 0xe41df53a
   name: "invert_bclk"
@@ -148428,6 +148522,17 @@ member {
   type_id: 0x6720d32f
   offset: 3136
 }
+member {
+  id: 0x4cc0da76
+  name: "leader_pulse"
+  type_id: 0x4585663f
+}
+member {
+  id: 0xb407cc56
+  name: "leader_space"
+  type_id: 0x4585663f
+  offset: 32
+}
 member {
   id: 0x2e4133fd
   name: "leaf"
@@ -162745,6 +162850,12 @@ member {
   type_id: 0xc9082b19
   offset: 7584
 }
+member {
+  id: 0x351e6705
+  name: "min_timeout"
+  type_id: 0xc9082b19
+  offset: 352
+}
 member {
   id: 0x351e6bb7
   name: "min_timeout"
@@ -165949,6 +166060,13 @@ member {
   type_id: 0x4585663f
   offset: 64
 }
+member {
+  id: 0xae927fdc
+  name: "msb_first"
+  type_id: 0x4585663f
+  offset: 224
+  bitsize: 1
+}
 member {
   id: 0x10b6405f
   name: "msb_right"
@@ -197841,6 +197959,12 @@ member {
   type_id: 0x2efa6d83
   offset: 192
 }
+member {
+  id: 0x5084eb47
+  name: "protocols"
+  type_id: 0x92233392
+  offset: 128
+}
 member {
   id: 0xa2bb1b42
   name: "protonum"
@@ -201635,6 +201759,12 @@ member {
   type_id: 0x578f9c2b
   offset: 16512
 }
+member {
+  id: 0xbc2dcf71
+  name: "raw_register"
+  type_id: 0x2d90ce2d
+  offset: 384
+}
 member {
   id: 0x3d617ac5
   name: "raw_request"
@@ -201726,6 +201856,12 @@ member {
   type_id: 0x295c7202
   offset: 1192
 }
+member {
+  id: 0x818075a5
+  name: "raw_unregister"
+  type_id: 0x2d90ce2d
+  offset: 448
+}
 member {
   id: 0x018e63c8
   name: "rawdescriptors"
@@ -243727,6 +243863,24 @@ member {
   type_id: 0x6720d32f
   offset: 256
 }
+member {
+  id: 0x0673a4c3
+  name: "trailer_pulse"
+  type_id: 0x4585663f
+  offset: 160
+}
+member {
+  id: 0x2a25d896
+  name: "trailer_space"
+  type_id: 0x4585663f
+  offset: 128
+}
+member {
+  id: 0x2a25d99d
+  name: "trailer_space"
+  type_id: 0x4585663f
+  offset: 192
+}
 member {
   id: 0xf5debb96
   name: "trans"
@@ -317796,6 +317950,50 @@ struct_union {
     member_id: 0x5c802411
   }
 }
+struct_union {
+  id: 0xc591d8f8
+  kind: STRUCT
+  name: "ir_raw_handler"
+  definition {
+    bytesize: 64
+    member_id: 0x7c00ef52
+    member_id: 0x5084eb47
+    member_id: 0x6e88de7a
+    member_id: 0xf632952b
+    member_id: 0xb623a0ae
+    member_id: 0x351e6705
+    member_id: 0xbc2dcf71
+    member_id: 0x818075a5
+  }
+}
+struct_union {
+  id: 0xd9c50754
+  kind: STRUCT
+  name: "ir_raw_timings_manchester"
+  definition {
+    bytesize: 20
+    member_id: 0x4cc0da76
+    member_id: 0xb407cc56
+    member_id: 0x458b8235
+    member_id: 0xeac27b10
+    member_id: 0x2a25d896
+  }
+}
+struct_union {
+  id: 0x44eb78eb
+  kind: STRUCT
+  name: "ir_raw_timings_pd"
+  definition {
+    bytesize: 32
+    member_id: 0xe8a40f03
+    member_id: 0xd9e8634f
+    member_id: 0xd8ab47be
+    member_id: 0xb6979874
+    member_id: 0x0673a4c3
+    member_id: 0x2a25d99d
+    member_id: 0xae927fdc
+  }
+}
 struct_union {
   id: 0xf1be35e6
   kind: STRUCT
@@ -389518,6 +389716,11 @@ function {
   parameter_id: 0x397d00ab
   parameter_id: 0x292cb268
 }
+function {
+  id: 0x1e654275
+  return_type_id: 0x48b5725f
+  parameter_id: 0x3bf44f99
+}
 function {
   id: 0x1e6686a2
   return_type_id: 0x48b5725f
@@ -399445,6 +399648,14 @@ function {
   parameter_id: 0x6720d32f
   parameter_id: 0x0b553223
 }
+function {
+  id: 0x8a043dfc
+  return_type_id: 0x6720d32f
+  parameter_id: 0x52a8f1d9
+  parameter_id: 0xc9082b19
+  parameter_id: 0x2f5073a5
+  parameter_id: 0x4585663f
+}
 function {
   id: 0x8a129bb5
   return_type_id: 0x19c40476
@@ -405829,6 +406040,11 @@ function {
   parameter_id: 0x3c5cda3d
   parameter_id: 0x31b5a66f
 }
+function {
+  id: 0x937df0c9
+  return_type_id: 0x6720d32f
+  parameter_id: 0x3bf44f99
+}
 function {
   id: 0x937fa102
   return_type_id: 0x6720d32f
@@ -422549,6 +422765,15 @@ function {
   parameter_id: 0x18bd6530
   parameter_id: 0x18bd6530
 }
+function {
+  id: 0x9cc8bbc4
+  return_type_id: 0x6720d32f
+  parameter_id: 0x0144254e
+  parameter_id: 0x4585663f
+  parameter_id: 0x39daf541
+  parameter_id: 0x4585663f
+  parameter_id: 0x92233392
+}
 function {
   id: 0x9cc9d859
   return_type_id: 0x6720d32f
@@ -422577,6 +422802,15 @@ function {
   parameter_id: 0x6720d32f
   parameter_id: 0x342a1515
 }
+function {
+  id: 0x9ccacf7d
+  return_type_id: 0x6720d32f
+  parameter_id: 0x0144254e
+  parameter_id: 0x4585663f
+  parameter_id: 0x300812ba
+  parameter_id: 0x4585663f
+  parameter_id: 0x92233392
+}
 function {
   id: 0x9ccae70d
   return_type_id: 0x6720d32f
@@ -425948,6 +426182,12 @@ function {
   return_type_id: 0x6720d32f
   parameter_id: 0x0f78474f
 }
+function {
+  id: 0x9e5fda8b
+  return_type_id: 0x6720d32f
+  parameter_id: 0x060cf413
+  parameter_id: 0x97012809
+}
 function {
   id: 0x9e5fde3c
   return_type_id: 0x6720d32f
@@ -480318,6 +480558,15 @@ elf_symbol {
   type_id: 0x1a42044c
   full_name: "hdmi_avi_infoframe_pack_only"
 }
+elf_symbol {
+  id: 0x668cde28
+  name: "hdmi_drm_infoframe_check"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x626713af
+  type_id: 0x9f3d453e
+  full_name: "hdmi_drm_infoframe_check"
+}
 elf_symbol {
   id: 0x7ec10e18
   name: "hdmi_drm_infoframe_init"
@@ -484590,6 +484839,42 @@ elf_symbol {
   type_id: 0x9db62354
   full_name: "ir_raw_event_store_edge"
 }
+elf_symbol {
+  id: 0xe481ba66
+  name: "ir_raw_gen_manchester"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xdbe61105
+  type_id: 0x9ccacf7d
+  full_name: "ir_raw_gen_manchester"
+}
+elf_symbol {
+  id: 0x6a48ce69
+  name: "ir_raw_gen_pd"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x9fcbee9f
+  type_id: 0x9cc8bbc4
+  full_name: "ir_raw_gen_pd"
+}
+elf_symbol {
+  id: 0x6d6a0985
+  name: "ir_raw_handler_register"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x65e3e661
+  type_id: 0x937df0c9
+  full_name: "ir_raw_handler_register"
+}
+elf_symbol {
+  id: 0xa01685ea
+  name: "ir_raw_handler_unregister"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x903a5fe0
+  type_id: 0x1e654275
+  full_name: "ir_raw_handler_unregister"
+}
 elf_symbol {
   id: 0xbf9d3356
   name: "irq_check_status_bit"
@@ -531089,6 +531374,7 @@ interface {
   symbol_id: 0x306e3b3d
   symbol_id: 0x684435da
   symbol_id: 0x4e8277c6
+  symbol_id: 0x668cde28
   symbol_id: 0x7ec10e18
   symbol_id: 0xe197ec6f
   symbol_id: 0x58069299
@@ -531563,6 +531849,10 @@ interface {
   symbol_id: 0x5ba96acf
   symbol_id: 0xd15ab841
   symbol_id: 0x915bc321
+  symbol_id: 0xe481ba66
+  symbol_id: 0x6a48ce69
+  symbol_id: 0x6d6a0985
+  symbol_id: 0xa01685ea
   symbol_id: 0xbf9d3356
   symbol_id: 0xd8989bc0
   symbol_id: 0x43c80a09
diff --git a/gki/aarch64/symbols/allwinner b/gki/aarch64/symbols/allwinner
index dd7dc449b3e7..07e268a6a7da 100644
--- a/gki/aarch64/symbols/allwinner
+++ b/gki/aarch64/symbols/allwinner
@@ -89,3 +89,8 @@
   pinctrl_gpio_set_config
   skb_append
   no_seek_end_llseek
+  ir_raw_handler_register
+  ir_raw_handler_unregister
+  ir_raw_gen_pd
+  ir_raw_gen_manchester
+  hdmi_drm_infoframe_check
-- 
2.50.1

